medcat-service/medcat_service/main.py

```diff
@@ -1,28 +1,56 @@
+import logging
+from contextlib import asynccontextmanager
+
 import gradio as gr
 from fastapi import FastAPI, Request
 from fastapi.responses import JSONResponse
 
+from medcat_service.config import Settings
 from medcat_service.demo.gradio_demo import io
-from medcat_service.dependencies import get_settings
+from medcat_service.dependencies import set_global_processor
+from medcat_service.nlp_processor.medcat_processor import MedCatProcessor
 from medcat_service.routers import admin, health, process
 from medcat_service.types import HealthCheckFailedException
 
-settings = get_settings()
 
-app = FastAPI(
-    title="MedCAT Service",
-    summary="MedCAT Service",
-    contact={
+@asynccontextmanager
+async def lifespan(app: FastAPI):
+
+    log = logging.getLogger(__name__)
+    log.debug("Starting MedCAT Service lifespan setup")
+
+    # allow overriding settings and medcat processor for testing
+    settings = getattr(app.state, "settings", None)
+    if settings is None:
+        settings = Settings()
+        app.state.settings = settings
```

Comment:
This is duplicated on line 33. Should be able to remove this line here and just keep the one after as far as I can tell.